specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json

```diff
@@ -1764,6 +1764,115 @@
+        "description": "Enable interactive authoring of Managed Virtual Network integration runtime.",
+        "x-ms-long-running-operation": true,
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/factoryName"
+          },
+          {
+            "$ref": "#/parameters/integrationRuntimeName"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "name": "enableInteractiveQueryRequest",
+            "description": "The enable interactive authoring integration runtime properties.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/EnableInteractiveQueryRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/IntegrationRuntimeResource"
+            }
+          },
+          "202": {
+            "description": "Accepted.",
+            "headers": {
+              "location": {
+                "description": "URI to poll for asynchronous operation status.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "An error response received from the Azure Data Factory service.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v6/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataFactory/factories/{factoryName}/integrationRuntimes/{integrationRuntimeName}/disableInteractiveQuery": {
+      "post": {
+        "tags": [
+          "integrationRuntimeDisableInteractiveQuery"
+        ],
+        "operationId": "IntegrationRuntime_DisableInteractiveQuery",
+        "x-ms-examples": {
+          "IntegrationRuntime_DisableInteractiveQuery": {
+            "$ref": "./examples/IntegrationRuntimes_DisableInteractiveQuery.json"
+          }
+        },
+        "description": "Disable interactive authoring of Managed Virtual Network integration runtime.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/factoryName"
+          },
+          {
+            "$ref": "#/parameters/integrationRuntimeName"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/IntegrationRuntimeResource"
```

Comment:
Actually, I feel like as a best practice I should maybe discourage you from returning the IntegrationRuntimeResource as the response - unless there's an actual reason to?

Resource states are readable through GET/SET responses for the most part - to also expose via POST API responses is sort of like a 'loophoole' for reading resource state, where someone might think they _just_ gave someone permission to post an action to a resource... and then finds out later (surprise!) they also exposed the resource state.

Reply:
How are they going to know to get a new sdk version  etc